EPRN.Portal/Controllers/HomeController.cs

```diff
@@ -1,21 +1,27 @@
-﻿using EPRN.Portal.ViewModels;
+﻿using EPRN.Common.Enums;
+using EPRN.Portal.Helpers.Interfaces;
+using EPRN.Portal.Services.Interfaces;
 using Microsoft.AspNetCore.Mvc;
 
 namespace EPRN.Portal.Controllers
 {
     public class HomeController : Controller
     {
-        public HomeController(ILogger<HomeController> logger)
+        public UserRole UserRole { get; set; }
+        private readonly IHomeServiceFactory _homeServiceFactory;
+        private IHomeService _homeService;
+
+        public HomeController(IHomeServiceFactory homeServiceFactory)
         {
+            _homeServiceFactory = homeServiceFactory ?? throw new ArgumentNullException(nameof(homeServiceFactory));
```

Comment:
I don't think the factory needs to be a private meber variable, does it? It's just being used in the constructor to retrieve the relevant service.